app/src/main/kotlin/com/simplemobiletools/calculator/helpers/CalculatorImpl.kt

```diff
@@ -28,21 +28,19 @@ import com.simplemobiletools.calculator.helpers.CONSTANT.TANGENT
 import java.io.File
 
 class CalculatorImpl(calculator: Calculator, private val context: Context) {
-    var displayedNumber: String? = null
-    var displayedFormula: String? = null
-    var lastKey: String? = null
-    private var mLastOperation: String? = null
+    var displayedFormula: String
```

Comment:
We have to ensure that these variables do not change because they will be used the saving of files. Saving the results and equations